src/status_im/ui/screens/status/views.cljs

```diff
@@ -61,44 +64,75 @@
-       [photos/member-identicon identicon])]
-    [react/view {:flex 1}
-     [react/view {:flex-direction  :row
-                  :justify-content :space-between}
+(defn on-long-press-fn [on-long-press content]
+  (on-long-press
+   [{:on-press #(react/copy-to-clipboard
+                 (components.reply/get-quoted-text-with-mentions
+                  (get content :parsed-text)))
+     :label    (i18n/label :t/sharing-copy-to-clipboard)}]))
+
+(defn message-item [timeline? account]
+  (fn [{:keys [content-type content from timestamp identicon outgoing] :as message} {:keys [modal on-long-press]}]
+    [react/view (merge {:padding-vertical   8
+                        :flex-direction     :row
+                        :background-color   (when (and timeline? outgoing) colors/blue-light)
+                        :padding-horizontal 16}
+                       (when modal
+                         {:border-radius 16
+                          :background-color colors/white}))
+     [react/view {:padding-top 2 :padding-right 8}
       (if outgoing
-        [message/message-my-name {:profile? true :you? false}]
-        [message/message-author-name from {:profile? true}])
-      [react/text {:style {:font-size 10 :color colors/gray}}
-       (datetime/time-ago (datetime/to-date timestamp))]]
-     (if (= content-type constants/content-type-image)
-       [image-message message]
-       [react/view
-        [message/render-parsed-text (assoc message :outgoing false) (:parsed-text content)]
-        [link-preview/link-preview-wrapper (:links content) outgoing]])]]])
+        [photos/member-identicon (multiaccounts/displayed-photo account)]
+        [photos/member-identicon identicon])]
+     [react/view {:flex 1}
+      [react/view {:flex-direction  :row
+                   :justify-content :space-between}
+       (if outgoing
+         [message/message-my-name {:profile? true :you? false}]
+         [message/message-author-name from {:profile? true}])
+       [react/text {:style {:font-size 10 :color colors/gray}}
+        (datetime/time-ago (datetime/to-date timestamp))]]
+      [react/touchable-highlight (when-not modal
+                                   {:on-long-press #(on-long-press-fn on-long-press content)})
+       [react/view (when modal {:background-color colors/white})
+        (if (= content-type constants/content-type-image)
+          [image-message message]
+          [message/render-parsed-text (assoc message :outgoing false) (:parsed-text content)])
+        [link-preview/link-preview-wrapper (:links content) outgoing]]]]]))
 
 (defn render-message [timeline? account]
   (fn [{:keys [type] :as message} idx]
-    (if (= type :datemark)
-      nil
-      (if (= type :gap)
-        (if timeline?
-          nil
-          [gap/gap message idx messages-list-ref])
-        ; message content
-        [message-item message timeline? account]))))
+    [(fn []
+       (if (= type :datemark)
```

Comment:
This looks like it could be inversed in a `when-not` to reduce the number of lines by removing nil's and identation to right will also reduce a bit.